### DISCUSSION TITLE
MACOS CFclient "Cannot find a Crazyradio Dongle"

### DISCUSSION THREAD

**Discussion by samuelensl** [bitcraze/discussions#1092] [Q&A]
Hi,

I'm starting a PHD project on the Crazyflie bolt drone and I'm having problems trying to connect the CFclient to the drone with a Crazyradio. More precisely, I use a python virtual environment on my MACOS (intel version) and I encounter problems when starting the CF client: "Cannot find a Carzyradio Dongle". Do you have any idea to solve this problem ?

Thanks

<img width="1283" alt="Capture d’écran 2023-12-14 à 16 50 44" src="https://github.com/bitcraze/discussions/assets/62340086/10910ea3-ec5f-42b8-8630-9582475950eb">

**Comment by knmcguire** [comment]
Hi!  Do you have a crazyradio2 by any chance? Then you first need to flash the new firmware: https://www.bitcraze.io/documentation/tutorials/getting-started-with-crazyradio-2-0/

If you did that already, could you check with your mac if it sees the crazyradio at all?

**Reply by samuelensl** [reply]
Thank you for your reactivity !
Yes, I have a Crazyradio 2.0. I tried to flash the new firmware by putting the Crazyradio in bootloader mode (led flashing red) but when I drag and drop the .uf2 file onto the Crazyradio it gives me an error message :
<img width="395" alt="Screenshot 2023-12-15 at 10 17 55" src="https://github.com/bitcraze/discussions/assets/62340086/cd54f842-3fc0-405b-afb1-a3cd5a2b0dc4">

**Reply by ataffanel** [reply]
This is not something we have ever seen before, what computer are you running exactly? What version of the operating system are you using?

Could you try on another computer to copy the firmware? It seems that your computer is not recognizing the boot-loader's virtual disk drive correctly which is causing the problem.

**Reply by j-makwana** [reply]
i got the same problem

**Reply by samuelensl** [reply]
The only solution that I find is to use a computer with windows OS and to use the Zadig software as described here : 

https://www.bitcraze.io/documentation/repository/crazyradio-firmware/master/building/usbwindows/
